mm/shm/shmat.c

```diff
@@ -32,11 +32,43 @@
 #include <nuttx/sched.h>
 #include <nuttx/arch.h>
 #include <nuttx/pgalloc.h>
+#include <nuttx/mm/map.h>
 
 #include "shm/shm.h"
 
 #ifdef CONFIG_MM_SHM
 
+/****************************************************************************
+ * Private Functions
+ ****************************************************************************/
+
+static int munmap_shm(FAR struct task_group_s *group,
+                      FAR struct mm_map_entry_s *map,
+                      FAR void *start,
+                      size_t length)
+{
+  int ret;
+
+  if (group)
+    {
+      /* Normal "unmap", let the shmdt do the work */
+
+      ret = shmdt(start);
+    }
+  else
+    {
+      /* Address environment is being deleted, and the group is not current.
+       * Get shmid from map->priv and only free the pages, don't touch any
+       * mappings
+       */
+
+      shm_destroy(map->priv.i);
```

Comment:
Here the whole attach counter logic was lost, need to still add that. It was not intended to destroy directly..

Reply:
fixed